pilot/pkg/xds/ads.go

```diff
@@ -294,6 +315,9 @@ func (s *DiscoveryServer) shouldRespond(con *Connection, request *discovery.Disc
 		if s.InternalGen != nil {
 			s.InternalGen.OnNack(con.proxy, request)
 		}
+		con.proxy.Lock()
+		con.proxy.WatchedResources[request.TypeUrl].NonceNacked = request.ResponseNonce
```

Comment:
Super minor nit  My only comment is this NACK is reset after an ACK. There is a comment somewhere else that explains that, but that comment may be better copied here, or in some way duplicated with different wording.